File: array/2053_kth_distinct_string_in_an_array/2053_kth_distinct_string_in_an_array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        tmp = {}
        for i in arr:
            if i not in tmp:
                tmp[i] = 1
            else:
                tmp[i] += 1
        
        for i in arr:
            if tmp[i] == 1:
                k -= 1
            if k == 0:
                return i
        return ""

File: array/2053_kth_distinct_string_in_an_array/2053_kth_distinct_string_in_an_array.go
package main

func kthDistinct(arr []string, k int) string {
	tmp := make(map[string]int, len(arr))
	for i := range arr {
		tmp[arr[i]] += 1
	}
	for i := range arr {
		if tmp[arr[i]] == 1 {
			k -= 1
		}
		if k == 0 {
			return arr[i]
		}

	}
	return ""

}

